src/util/payee-name.js

```diff
@@ -0,0 +1,42 @@
+import { title } from './title/index.js';
+
+function formatPayeeIban(iban) {
+  return '(' + iban.slice(0, 4) + ' XXX ' + iban.slice(-4) + ')';
+}
+
+export const formatPayeeName = (trans) => {
+  const nameParts = [];
+
+  // get the correct name and account fields for the transaction amount
+  let name;
+  let account;
+  if (trans.amount > 0 || Object.is(Number(trans.amount), 0)) {
+    name = trans.debtorName;
+    account = trans.debtorAccount;
+  } else {
+    name = trans.creditorName;
+    account = trans.creditorAccount;
+  }
+
+  // use the correct fields for the amount if it was found
+  // if not, use whatever we can find
+  name =
+    name ||
+    trans.debtorName ||
+    trans.creditorName ||
+    trans.remittanceInformationUnstructured ||
+    (trans.remittanceInformationUnstructuredArray || []).join(', ') ||
+    trans.additionalInformation;
```

Comment:
Since the if/else above sets `name` to something (possibly), you can avoid the assignment here and simplify it quite a bit in that case it found a name already.

```suggestion
  if (!name) {
    name =
      trans.remittanceInformationUnstructured ||
      (trans.remittanceInformationUnstructuredArray || []).join(', ') ||
      trans.additionalInformation;
  }
```

Reply:
This changes the logic, the key functional part of this change is to allow creditorName to be used even if GoCardless /should/ be using debtorName instead and vice versa.

Happy to put the whole assignment into an if and drop the first `name || ` line if that aligns more closely with the style in this project?

Reply:
That is curious logic!  I guess if that is what needs to happen for GoCardless, the existing code is probably fine.  I am not sure if there is a style preference between using an `if` to avoid `name = name` or leave it as is.